lib/widgets/pokemon_tile_item.dart

```diff
@@ -1,14 +1,23 @@
 import 'package:flutter/material.dart';
 import 'package:go_router/go_router.dart';
-import 'package:pokedex/utils/strings.dart' as str;
+import 'package:pokedex/api/pokedex_api_spec/model/pokemon.dart';
+import 'package:pokedex/utils/constant.dart' as k;
+import 'package:dartx/dartx.dart';
 
 class PokemonTileItem extends StatelessWidget {
-  const PokemonTileItem({super.key});
+  const PokemonTileItem({
+    super.key,
+    required this.pokemon,
+    required this.pokemonImg,
+  });
+
+  final Pokemon pokemon;
+  final String pokemonImg;
 
   @override
   Widget build(BuildContext context) {
     return GestureDetector(
-      onTap: () => context.pop(),
+      onTap: () => context.go('/details', extra: pokemon),
```

Comment:
this is already for details page right, which is unrelated to this pr? have it `onTap: () => {},` instead